src/magneticField/CMZField.cpp

```diff
@@ -0,0 +1,441 @@
+    b.x +=-sin(pphi)*(-BrAz(rr,pphi,zz,0,B1,1.,R,R/10.));
+    b.y +=cos(pphi)*(-BrAz(rr,pphi,zz,0,B1,1.,R,R/10.)); 
+    return b*gauss;
+} 
+Vector3d CMZField::getICField(const Vector3d& pos) const {//Field in intercloud medium--> poloidal field
+    Vector3d b(0.);
+    //poloidal field in the intercloud region
+    double x=pos.x;
+    double y=pos.y -(-8.3*pc);
+    double z=pos.z -(-6.9*pc);
+    double a2=2.*158.*pc;// Radius of A
+    double a1=2.*35.*pc;
+    double eta=0.85;
+    double B1=1e-5*gauss;
+    double B2 = B1/eta;
+
+    double r = sqrt(x*x+y*y);
+    double phi = std::atan2(y,x);
+    double sPhi = sin(phi);
+    double cPhi = cos(phi);
+    double a = 4*log(2)/a1/a1;
+    double L = a2/log(2)/2.;
+    double r1 = 1/(1+a*z*z);
+    double Br =2*a*r*pow(r1,3)*z*B2*exp(-r/L*r1);
+
+    b.y += sPhi*Br;
+    b.x += cPhi*Br;
+    b.z += r1*r1 * B2 * exp(-r/L *r1);
+    return b;
+}                                                         
+    //  
+Vector3d CMZField::getNTFField(const Vector3d& pos) const {//Field in the non-thermal filaments--> predominantly poloidal field
+    Vector3d b(0.);
+    //poloidal field in the non-thermal filament region (except "pelical"-> azimuthal)
+    double arcmin=1./60.;
+
+    //A=SgrC
+    double y_mid=-81.59*pc;//  y midpoint of A
+    double z_mid=-1.48*pc;//  z midpoint of A
+    double xx=pos.x;
+    double yy=pos.y-y_mid;// shifted coordinates
+    double zz=pos.z-z_mid;// shiftes coordinates
+    double a1=27.44*pc;// arcmin-> deg->cm
+    double a2=1.73*pc;// arcmin-> deg-> cm
+    double eta=0.48;
+    double B1=1.e-4;
+    b.y+=ByPol(xx,yy,zz,B1,a1,a2,eta);
+    b.x+=BxPol(xx,yy,zz,B1,a1,a2,eta);
+    b.z+=BzPol(xx,yy,zz,B1,a1,a2,eta);
+
+    //A=G359.15-0.2 The Snake
+    y_mid=-126.1*pc;//  y midpoint of A
+    z_mid=-25.22*pc;//  z midpoint of A
+    yy=pos.y-y_mid;// shifted coordinates
+    zz=pos.z-z_mid;// shiftes coordinates
+    a1=12.86*pc;// arcmin-> deg->cm
+    a2=2.22*pc;// arcmin-> deg-> cm
+    B1=88.e-6;
+    b.y+=ByPol(xx,yy,zz,B1,a1,a2,eta);
+    b.x+=BxPol(xx,yy,zz,B1,a1,a2,eta);
+    b.z+=BzPol(xx,yy,zz,B1,a1,a2,eta);
+
+    //A=G359.54+0.18 Nonthermal Filament
+    y_mid=-68.24*pc;//  y midpoint of A
+    z_mid=25.22*pc;//  z midpoint of A
+    yy=pos.y-y_mid;// shifted coordinates
+    zz=pos.z-z_mid;// shiftes coordinates
+    a1=15.08*pc;// arcmin-> deg->cm
+    a2=2.72;// arcmin-> deg-> cm
+    B1=1.e-3;
+    b.y+=ByPol(xx,yy,zz,B1,a1,a2,eta);
+    b.x+=BxPol(xx,yy,zz,B1,a1,a2,eta);
+    b.z+=BzPol(xx,yy,zz,B1,a1,a2,eta);
+
+    //A=G359.79 +17 Nonthermal Filament
+    y_mid=-31.15*pc;//  y midpoint of A
+    z_mid=23.74*pc;//  z midpoint of A
+    yy=pos.y-y_mid;// shifted coordinates
+    zz=pos.z-z_mid;// shiftes coordinates
+    a1=16.07*pc;// arcmin-> deg->cm
+    a2=3.46*pc;// arcmin-> deg-> cm
+    B1=1.e-3;
+    b.y+=ByPol(xx,yy,zz,B1,a1,a2,eta);
+    b.x+=BxPol(xx,yy,zz,B1,a1,a2,eta);
+    b.z+=BzPol(xx,yy,zz,B1,a1,a2,eta);
+
```

Comment:
Why is that block commented out?

Reply:
This is an older part of the code. G359.91 is only a possible Nonthermal Filament but might be a SNR. 
I have removed this part at this point. Interessed users could add it for themselve.